sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/memory.scala

```diff
@@ -137,7 +140,7 @@ class MemoryStreamScanBuilder(stream: MemoryStreamBase[_]) extends ScanBuilder w
  * is intended for use in unit tests as it can only replay data when the object is still
  * available.
  */
-case class MemoryStream[A : Encoder](id: Int, sqlContext: SQLContext)
+case class MemoryStream[A : Encoder](id: Int, sqlContext: SQLContext, numPartitions: Int = 1)
```

Comment:
I am little concerned about changing the default behavior of this class. Earlier the default behavior was to put each "new block" into a partition (i.e., each new seq of data added through `addData` formed a partition). So in a unit test, if there were multiple `AddData` made before starting a stream, the first batch could have multiple partitions. With this change, there will be always 1 partition. I am not sure if there are any unit test requires this behavior to be correct, but I am a little afraid of having an unpredictable impact on other tests. So I would make this param optional to None (that is, no fixed number of partitions), and preserve the existing behavior.

Reply:
I think I missed the intention of the code, my bad. I thought there's always only 1 partition (not sure why I missed that, fairly old code for me as well, sorry). We can set this to None by default and conditionally applies the change. Thanks for the correction.